plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.cpp

```diff
@@ -2126,98 +2022,17 @@ bool ProxyShape::isBounded() const
-  MDataHandle outDataHandle = dataBlock.inputValue(m_outStageData, &status);
-  (void)outDataHandle;
-  CHECK_MSTATUS_AND_RETURN(status, MBoundingBox() );
-
-  // XXX:aluk
-  // If we could cheaply determine whether a stage only has static geometry,
-  // we could make this value a constant one for that case, avoiding the
-  // memory overhead of a cache entry per frame
-  UsdTimeCode currTime = UsdTimeCode(inputTimeValue(dataBlock, m_outTime).as(MTime::uiUnit()));
-
-  // RB: There must be a nicer way of doing this that avoids the map?
-  // The time codes are likely to be ranged, so an ordered array + binary search would surely work?
-  std::map<UsdTimeCode, MBoundingBox>::const_iterator cacheLookup = m_boundingBoxCache.find(currTime);
-  if (cacheLookup != m_boundingBoxCache.end())
-  {
-    return cacheLookup->second;
-  }
-
-  GfBBox3d allBox;
-  UsdPrim prim = getUsdPrim(dataBlock);
-  if (prim)
-  {
-    UsdGeomImageable imageablePrim(prim);
-    bool showGuides = inputBoolValue(dataBlock, m_displayGuides);
-    bool showRenderGuides = inputBoolValue(dataBlock, m_displayRenderGuides);
-    if (showGuides && showRenderGuides)
-    {
-      allBox = imageablePrim.ComputeUntransformedBound(
-            currTime,
-            UsdGeomTokens->default_,
-            UsdGeomTokens->proxy,
-            UsdGeomTokens->guide,
-            UsdGeomTokens->render);
-    }
-    else
-    if (showGuides && !showRenderGuides)
-    {
-      allBox = imageablePrim.ComputeUntransformedBound(
-            currTime,
-            UsdGeomTokens->default_,
-            UsdGeomTokens->proxy,
-            UsdGeomTokens->guide);
-    }
-    else if (!showGuides && showRenderGuides)
-    {
-      allBox = imageablePrim.ComputeUntransformedBound(
-            currTime,
-            UsdGeomTokens->default_,
-            UsdGeomTokens->proxy,
-            UsdGeomTokens->render);
-    }
-    else
-    {
-      allBox = imageablePrim.ComputeUntransformedBound(
-            currTime,
-            UsdGeomTokens->default_,
-            UsdGeomTokens->proxy);
-    }
-  }
-  else
-  {
-    return MBoundingBox();
-  }
-
-  // insert new cache entry
-  MBoundingBox& retval = m_boundingBoxCache[currTime];
-
-  // Convert to GfRange3d to MBoundingBox
-  GfRange3d boxRange = allBox.ComputeAlignedBox();
-  if (!boxRange.IsEmpty())
-  {
-    retval = MBoundingBox(MPoint(boxRange.GetMin()[0],
-                                 boxRange.GetMin()[1],
-                                 boxRange.GetMin()[2]),
-                          MPoint(boxRange.GetMax()[0],
-                                 boxRange.GetMax()[1],
-                                 boxRange.GetMax()[2]));
-  }
-  else
-  {
-    retval = MBoundingBox(MPoint(-100000.0f, -100000.0f, -100000.0f), MPoint(100000.0f, 100000.0f, 100000.0f));
-  }
+  cachedBBox = MBoundingBox(
+      MPoint(-100000.0f, -100000.0f, -100000.0f),
+      MPoint( 100000.0f,  100000.0f,  100000.0f));
```

Comment:
This is a bug. Please evaluate the correct bounding box!!! reasons are:
1. We have city sized sets. This bounding box is far too small for that case. 
2. We have lego sized bricks, This bounding box is far too big for that case. 
3. Selection speed is already painfully slow for us. Attempting to force every proxy shape into a selection query won't be great for performance. :( 
4. We need the bounding box to remain accurate for animated objects.

Reply:
See the method name - ProxyShape::CacheEmptyBoundingBox. The calculation is now provided by MayaUsdProxyShapeBase::boundingBox()

Reply:
Ahh, ok. Ignore me then :)